specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json

```diff
@@ -720,6 +736,25 @@
           "description": "The name of SKU",
           "readOnly": true
         },
+        "billingPlans": {
+          "type": "array",
+          "description": "The billing plan options available for this SKU.",
+          "items" :{
+            "type": "object", 
+            "properties": {
+              "name": {
+                "$ref": "#/definitions/ReservationTerm",
+                "description": "The term for the billing SKU is available for."
+              }
+            },
+            "additionalProperties": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/ReservationBillingPlan"
+              }
+            }
```

Comment:
This is basically declaring that this object allows arbitrary array properties to be specified. Are these known ahead of time. Is there a known schema?

Reply:
Also what does the property name represent here?

---
In reply to: [325447456](https://github.com/Azure/azure-rest-api-specs/pull/7206#discussion_r325447456) [](ancestors = 325447456)

Reply:
billingPlans contains a array of objects, these objects names can be values from the enum #/definitions/ReservationBillingPlan. E.g. P1Y, P3Y
Each of these objects contains an array that can contain the values from the enum #/definitions/ReservationBillingPlan. E.g. Monthly, Upfront.

In the example for this api this looks like:
"billingPlans": [{
              "P1Y": [
                "Upfront",
                "Monthly"
              ]
            }, {
              "P3Y": [
                "Upfront",
                "Monthly"
              ]
            }
          ]

Reply:
Sounds good.